rest/tests/test_rest.py

```diff
@@ -0,0 +1,150 @@
+"""
+Unit tests for REST api
+"""
+
+from __future__ import unicode_literals
+
+import json
+
+from learningresources.tests.base import LoreTestCase
+from taxonomy.api import create_vocabulary
+from taxonomy.models import Vocabulary
+from rest.serializers import VocabularySerializer
+
+HTTP_OK = 200
+HTTP_CREATED = 201
+HTTP_NO_CONTENT = 204
+HTTP_UNAUTHORIZED = 403
+
+
+class TestRest(LoreTestCase):
+    """
+    REST test
+    """
+
+    def test_vocabulary(self):
+        """
+        Test for Vocabulary
+        """
+
+        resp = self.client.get('/api/v1/vocabularies/')
+        self.assertEqual(HTTP_OK, resp.status_code)
+        vocabularies = json.loads(resp.content.decode('utf-8'))
+        self.assertEqual(0, vocabularies['count'])
+
+        input_dict = {
+            'name': 'name',
+            'description': 'description',
+            'required': True,
+            'weight': 1000,
+            'vocabulary_type': 'f',
+            'repository': self.repo.id,
+        }
+        resp = self.client.post('/api/v1/vocabularies/', input_dict)
+        self.assertEqual(HTTP_CREATED, resp.status_code)
+        output_dict = json.loads(resp.content.decode('utf-8'))
+        self.assertEqual(input_dict['name'],
+                         output_dict['name'])
+
+        resp = self.client.get('/api/v1/vocabularies/')
+        self.assertEqual(HTTP_OK, resp.status_code)
+        vocabularies = json.loads(resp.content.decode('utf-8'))
+        self.assertEqual(1, vocabularies['count'])
+        new_id = vocabularies['results'][0]['id']
+
+        resp = self.client.get('/api/v1/vocabularies/{id}/'.format(
+            id=new_id
+        ))
+        self.assertEqual(HTTP_OK, resp.status_code)
+        output_dict = json.loads(resp.content.decode('utf-8'))
+        self.assertEqual(output_dict['name'], input_dict['name'])
+
+        resp = self.client.delete('/api/v1/vocabularies/{id}/'.format(
+            id=new_id
+        ))
+        self.assertEqual(HTTP_NO_CONTENT, resp.status_code)
+
+        resp = self.client.get('/api/v1/vocabularies/')
+        self.assertEqual(HTTP_OK, resp.status_code)
+        vocabularies = json.loads(resp.content.decode('utf-8'))
+        self.assertEqual(0, vocabularies['count'])
+
+    def test_term(self):
+        """Test REST access for term"""
+        vocab = create_vocabulary(
+            self.repo.id,
+            "name",
+            "description",
+            True,
+            Vocabulary.FREE_TAGGING,
+            1000
+        )
+        resp = self.client.get('/api/v1/terms/')
```

Comment:
They will change if the user changes the name of the vocabulary or repository, or the label on the term. I think this kind of access is restricted to curators so it may not be a big deal. Ideally URLs [never change](http://www.w3.org/Provider/Style/URI.html) but the alternative is slugs that may not match up with the name at all if the name changes, which may be confusing.